# Python for Data Science Assignment

### 1. Read the dataset to the python environment.

In [1]:
import pandas as pd

In [9]:
# Read the Iris dataset
iris = pd.read_csv('iris.csv')

In [10]:
iris.shape

(150, 5)

In [11]:
iris.head()

,SL,SW,PL,PW,Classification
0,5.1,3.5,1.4,0.2,Iris-setosa
1,4.9,3.0,1.4,0.2,Iris-setosa
2,4.7,3.2,1.3,0.2,Iris-setosa
3,4.6,3.1,1.5,0.2,Iris-setosa
4,5.0,3.6,1.4,0.2,Iris-setosa


### 2. Display the columns in the dataset

In [12]:
print('Columns in the dataset:')
iris.columns.tolist()

Columns in the dataset:


['SL', 'SW', 'PL', 'PW', 'Classification']

### 3. Calculate the mean of each column of the dataset.

In [13]:
# Calculate the mean of numeric columns
mean_values = iris.mean(numeric_only=True)

print('Mean of each numeric column:')
mean_values

Mean of each numeric column:


SL    5.843333
SW    3.054000
PL    3.758667
PW    1.198667
dtype: float64

### 4.Check for the null values present in the dataset.

In [17]:
print('Null values in each column:')
iris.isnull().sum()

Null values in each column:


SL                0
SW                0
PL                0
PW                0
Classification    0
dtype: int64

### 5.Meaningful Data Analysis / Visualizations

### 5.1 Number of Samples in Each Iris Class

This shows how many records belong to each classification.

In [18]:
print('Number of samples in each classification:')
iris['Classification'].value_counts()

Number of samples in each classification:


Classification
Iris-setosa        50
Iris-versicolor    50
Iris-virginica     50
Name: count, dtype: int64

### 5.2 Average Measurements for Each Iris Class

This compares the average Sepal Length, Sepal Width, Petal Length, and Petal Width for each classification.

In [19]:
print('Average measurements for each classification:')
iris.groupby('Classification')[['SL', 'SW', 'PL', 'PW']].mean()

Average measurements for each classification:


,SL,SW,PL,PW
Classification,,,,
Iris-setosa,5.006,3.418,1.464,0.244
Iris-versicolor,5.936,2.770,4.260,1.326
Iris-virginica,6.588,2.974,5.552,2.026


### 5.3 Minimum and Maximum Measurements

This shows the minimum and maximum values for each numeric measurement.

In [21]:
print('Minimum and maximum values:')
iris[['SL', 'SW', 'PL', 'PW']].agg(['min', 'max'])

Minimum and maximum values:


,SL,SW,PL,PW
min,4.3,2.0,1.0,0.1
max,7.9,4.4,6.9,2.5


## Conclusion

The Iris dataset contains 150 records and 5 columns. The dataset contains four numeric measurement columns (SL, SW, PL, PW) and one classification column. There are no null values. The Pandas analysis above helps compare the number of samples, average measurements, and measurement ranges for the Iris classifications.